bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -0,0 +1,265 @@
+
+                                    // listen to responses
+                                    Selector selector = Selector.open();
+                                    ByteBuffer buffer = ByteBuffer.wrap(new byte[50]);
+                                    channel.register(selector, SelectionKey.OP_READ);
+                                    selector.select(timeoutMs);
+                                    Iterator<SelectionKey> it = selector.selectedKeys().iterator();
+
+                                    switch (Objects.toString(response)) {
+                                        case ".*":
+                                            if (it.hasNext()) {
+                                                final SocketAddress source = ((DatagramChannel) it.next().channel())
+                                                        .receive(buffer);
+                                                logger.debug("Received return frame from {}",
+                                                        ((InetSocketAddress) source).getAddress().getHostAddress());
+                                                suggestions.add(candidate);
+                                                logger.debug("Suggested add-on found: {}", candidate.getUID());
+                                            } else {
+                                                logger.trace("{}: no response", candidate.getUID());
+                                            }
+                                            break;
+                                        default:
+                                            logger.info("{}: match-property response \"{}\" is unknown",
+                                                    candidate.getUID(), type);
+                                            break; // end loop
+                                    }
+
+                                } catch (IOException e) {
+                                    logger.debug("{}: network error", candidate.getUID(), e);
+                                }
+                            }
+                            break;
+
+                        default:
+                            logger.info("{}: match-property type \"{}\" is unknown", candidate.getUID(), type);
+                    }
+                } catch (ParseException | NumberFormatException none) {
+                    continue;
+                }
+            }
+        }
+    }
+
+    byte[] buildRequestArray(DatagramChannel channel, String request) throws java.io.IOException, ParseException {
+        InetSocketAddress sock = (InetSocketAddress) channel.getLocalAddress();
+
+        ByteArrayOutputStream requestFrame = new ByteArrayOutputStream();
+        StringTokenizer parts = new StringTokenizer(request);
+
+        while (parts.hasMoreTokens()) {
+            String token = parts.nextToken();
+            if (token.startsWith("$")) {
+                switch (token) {
+                    case "$src_ip":
```

Comment:
Should be a constant, and I would prefer camel case.

Reply:
It is just the hint that at this point of the frame the current IP needs to be inserted...